test/test_binary_ufuncs.py

```diff
@@ -654,12 +654,16 @@ def to_np(value):
                 actual = base.pow(exponent)
                 self.assertEqual(actual, expected.to(actual))
                 actual = base.clone()
-                if torch.can_cast(torch.result_type(base, exponent), base.dtype):
+                # When base is a 0-dim cpu tensor and exp is a cuda tensor, we exp `pow` to work but `pow_` to fail, since
+                # `pow` will try to create the output tensor on a cuda device, but `pow_` needs to use the cpu tensor as the output
+                if base.dim() == 0 and base.device.type == 'cpu' and exponent.device.type == 'cuda':
+                    regex = 'Expected all tensors to be on the same device, but found at least two devices, cuda.* and cpu!'
```

Comment:
This string isn't being used to generate a runtime exception.

Reply:
I'll add the following line after this one in another PR:
`self.assertRaisesRegex(RuntimeError, regex, torch.pow, base, exp)`

Reply:
Added `self.assertRaisesRegex(RuntimeError, regex, base.pow_, exponent)` if exponent is also a tensor.